opendevin/agent.py

```diff
@@ -46,24 +45,18 @@ class Agent(ABC):
     It tracks the execution status and maintains a history of interactions.
 
     :param instruction: The instruction for the agent to execute.
-    :param workspace_dir: The working directory for the agent.
     :param model_name: The litellm name of the model to use for the agent.
-    :param max_steps: The maximum number of steps to run the agent.
     """
 
     _registry: Dict[str, Type['Agent']] = {}
 
     def __init__(
         self,
         instruction: str,
-        workspace_dir: str,
-        model_name: str,
-        max_steps: int = 100
```

Comment:
I get rid of `workspace_dir` and `max_steps` since these are handled by the `Controller` now.